src/components/MovieDetailsPage.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import {connect} from 'react-redux'
+import { Redirect } from 'react-router-dom'
+
+import {showMovieDetails} from '../redux/actions'
+
+class MovieDetailsPage extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      backToSearch: false,
+    }
+    //fetch movie data
+    this.props.showMovieDetails(this.props.match.params.movieId)
+  }
+
+  render() {
+    var contentHTML
+    //redirect back to search
+    if (this.state.backToSearch === true) {
+      return <Redirect to= '/'/>
+    }
+    //alert on error
+    if (this.props.error.isError === true) {
+      contentHTML = <div class="error" align="center"> {this.props.error.message} </div>
+    }
+    //loading
+    else if (this.props.isLoading === true) {
+      const Loading = require('react-loading-animation');
+      contentHTML = <Loading align="center"/>
+    }
+    //show details
+    else {
+      contentHTML = (
+        <table>
+          <tr style={{padding: 18}}>
+            <td> <img src= {this.props.movieDetails.Poster} /> </td>
+            <td> {Object.keys(this.props.movieDetails).map(
+              property => this.showMovieInfo(property, this.props.movieDetails[property]))}
+            </td>
+          </tr>
+        </table>)
+    }
+
+     return (
+      <div align="center" class="padding">
+      {contentHTML}
+      <button onClick={() => this.goBackToSearchScreen()}> Back To Search </button>
+      </div>
+    )
+  }
+
+  //change state to route back to search
+  goBackToSearchScreen() {
+    this.setState({backToSearch: true,})
```

Comment:
Usually the way to navigate between screen is done differently.. in your flow:

1) you change the state
2) in the next render() you Redirect

Usually the proper way is to call some dispatch action (via some props, set in `mapDispatchToProps`) and responding in the Saga to the action and navigate from there. I just looked for some references and it seems like this can be done with `history` object. Can you please read the answers in 
https://stackoverflow.com/questions/42123261/programmatically-navigate-using-react-router-v4